core/src/test/scala/org/apache/spark/status/protobuf/KVStoreProtobufSerializerSuite.scala

```diff
@@ -31,6 +34,32 @@ import org.apache.spark.ui.scope.{RDDOperationEdge, RDDOperationNode}
 class KVStoreProtobufSerializerSuite extends SparkFunSuite {
   private val serializer = new KVStoreProtobufSerializer()
 
+  test("All the string fields must be optional to avoid NPE") {
+    val protoFile = getWorkspaceFilePath(
+      "core", "src", "main", "protobuf", "org", "apache", "spark", "status", "protobuf",
+      "store_types.proto")
+
+    val containsStringRegex = "\\s*string .*"
+    val invalidDefinition = new mutable.ArrayBuffer[(String, Int)]()
+    var lineNumber = 1
+    Source.fromFile(protoFile.toFile.getCanonicalPath).getLines().foreach { line =>
+      if (line.matches(containsStringRegex)) {
```

Comment:
`org.apache.spark.util.Utils.tryWithResource` should be used to ensure`Source.fromFile(protoFile.toFile.getCanonicalPath)` is closed

Reply:
+1 for the above comment.

Reply:
ok, updated